rumqttd/src/router/routing.rs

```diff
@@ -473,6 +486,14 @@ impl Router {
             for request in tracker.data_requests.iter_mut() {
                 if let Some(cursor) = retransmissions.get(&request.filter_idx) {
                     request.cursor = *cursor;
+                    // reset the group cursor
+                    if let Some(group_name) = &request.group {
+                        // TODO: Test this more
+                        self.shared_subscriptions
+                            .get_mut(group_name)
+                            .unwrap()
```

Comment:
We probably don't want to unwrap. Should be a way we can assert that it exists or atleast safely handle it.

Reply:
If there was a `DataRequest` created for group, we MUST have a entry in shared_subscriptions as well! We can also do
`self.shared_subscriptions[group_name]..` so we don't have to unwrap/check

Reply:
turns out, we can't use `[index]` to mutably borrow, so gotta settle with `get_mut()`